### PR TITLE
Adam optimizer

### PR DESCRIPTION
**Pull request by memimo**
Haven't tested the convergence rate yet. Just putting it here in case there is any comment about format or anyone catches a bug.

### CODE REVIEW

**Comment by memimo** [comment]
Seems to be converging more or less as expected. Tested on mnist with single layer fully connected model vs sgd.

File: blocks/algorithms/__init__.py [hunk 1]
```diff
@@ -534,3 +534,59 @@ def compute_steps(self, previous_steps):
...
+        self.epsilon = epsilon
+        self.decay_factor = decay_factor
+
+    def compute_step(self, param, previous_step):
+        mean = shared_floatx(param.get_value() * 0., 'mean')
+        variance = shared_floatx(param.get_value() * 0., 'variance')
+        time = shared_floatx(0., 'time')
+
+        t1 = time + 1
+        learning_rate = self.learning_rate * \
```
Comment (rizar) [inline blocks/algorithms/__init__.py]: The line breaks with "\" are discouraged in the Python style guide except for a few cases, so we in Blocks try not to use them. Can you replace those with parentheses?

Comment (memimo) [inline blocks/algorithms/__init__.py]: Sure, removed them.

**Comment by rizar** [comment]
I do not know much about this optimizer, but if you say it works, let's just merge it.

**Pull request merged**